alibi_detect/od/sklearn/base.py

```diff
@@ -0,0 +1,220 @@
+        """Abstract fit method.
+
+        Parameters
+        ----------
+        x
+            `torch.Tensor` to fit object on.
+        """
+        return self
+
+    def _set_fitted(self) -> Self:
+        """Sets the fitted attribute to True.
+
+        Should be called within the object fit method.
+        """
+        self.fitted = True
+        return self
+
+    def check_fitted(self):
+        """Checks to make sure object has been fitted.
+
+        Raises
+        ------
+        NotFittedError
+            Raised if method called and object has not been fit.
+        """
+        if not self.fitted:
+            raise NotFittedError(self.__class__.__name__)
+
+
+class SklearnOutlierDetector(FitMixinSklearn, ABC):
+    """Base class for sklearn backend outlier detection algorithms."""
+    threshold_inferred = False
+    threshold = None
+
+    def __init__(self):
+        super().__init__()
+
+    @abstractmethod
+    def score(self, x: np.ndarray) -> np.ndarray:
+        """Score the data.
+
+        Parameters
+        ----------
+        x
+            Data to score.
+
+        """
+        pass
+
+    def check_threshold_inferred(self):
+        """Check if threshold is inferred.
+
+        Raises
+        ------
+        ThresholdNotInferredError
+            Raised if threshold is not inferred.
+        """
+        if not self.threshold_inferred:
+            raise ThresholdNotInferredError(self.__class__.__name__)
+
+    @staticmethod
+    def _to_numpy(arg: Union[np.ndarray, SklearnOutlierDetectorOutput]) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+        """Map params to numpy arrays.
+
+        Note
+        ----
+        This function is for interface compatibility with the other backends. As such it does nothing but
+        return the input and unpack `SklearnOutlierDetectorOutput` into a `dict`.
+
+        Parameters
+        ----------
+        x
+            Data to convert.
+
+        Returns
+        -------
+        `np.ndarray` or dictionary of containing `numpy` arrays
+        """
+        if isinstance(arg, SklearnOutlierDetectorOutput):
+            return asdict(arg)
+        return arg
+
+    @staticmethod
+    def _to_tensor(x: Union[List, np.ndarray]) -> np.ndarray:
+        """Converts the data to a tensor.
+
+        This function is for interface compatibility with the other backends. As such it does nothing but
+        return the input.
+
+        Parameters
+        ----------
+        x
+            Data to convert.
+        """
+        return np.array(x)
+
+    def _classify_outlier(self, scores: np.ndarray) -> np.ndarray:
+        """Classify the data as outlier or not.
+
+        Parameters
+        ----------
+        scores
+            Scores to classify. Larger scores indicate more likely outliers.
+
+        Returns
+        -------
+        `np.ndarray` or ``None``
+        """
+        return scores > self.threshold if self.threshold_inferred else None
```

Comment:
This results in True/False detector outputs rather than 0/1 like other detectors